pkg/streamingpromql/query.go

```diff
@@ -310,10 +313,11 @@ func (q *Query) convertToInstantVectorOperator(expr parser.Expr, timeRange types
 }
 
 func (q *Query) convertFunctionCallToInstantVectorOperator(e *parser.Call, timeRange types.QueryTimeRange) (types.InstantVectorOperator, error) {
-	// Handle special toggles for classic histograms
-	if !q.engine.featureToggles.EnableHistogramQuantileFunction {
-		if e.Func.Name == "histogram_quantile" {
-			return nil, compat.NewNotSupportedError(fmt.Sprintf("'%s' function", e.Func.Name))
+	for _, fName := range q.engine.mqeOpts.DisabledFunctions {
+		// e.Func.Name is already validated by the parser. Meaning we don't need to check if the function name
+		// is real before checking if it is disabled.
+		if e.Func.Name == fName {
+			return nil, compat.NewNotSupportedError(fmt.Sprintf("'%s' function is disabled", e.Func.Name))
```

Comment:
In all other cases, we don't differentiate between a feature being unsupported entirely and a feature being disabled by a feature flag:

```suggestion
		if e.Func.Name == fName {
			return nil, compat.NewNotSupportedError(fmt.Sprintf("'%s' function", e.Func.Name))
```

Reply:
I feel like this is a little more explicitly disabled so the log line is helpful

Reply:
Not sure I follow sorry - isn't it just as explicit as any other feature disabled with a feature flag?

Reply:
The others are "enable" which default to `true`. This explicitly disables something, which is useful to know in the logs imo.

Reply:
I still think it's worthwhile, especially to differentiate between a function that is disabled verses one that we haven't got any support for (ie is missing from `instantVectorFunctionOperatorFactories`).

Not a hill I care for though if you want to veto this that's fine :-).

Reply:
At present, we don't differentiate between those cases, and I'd prefer to remain consistent with what we already have.